cmd/api/book/service.go

```diff
@@ -37,23 +37,50 @@ func (s *Service) ArchiveBook(id uuid.UUID) (Book, error) {
 	return s.repo.SetBookArchiveStatus(id, archived)
 }
 
-func (s *Service) CreateBook(bookEntry Book) (Book, error) {
-	bookEntry.ID = uuid.New()                                      //Atribute an ID to the entry
-	bookEntry.CreatedAt = time.Now().UTC().Round(time.Millisecond) //Atribute creating and updating time to the new entry. UpdateAt can change later.
-	bookEntry.UpdatedAt = bookEntry.CreatedAt
-	return s.repo.CreateBook(bookEntry)
+type EntryBookRequest struct {
+	Name      string
+	Price     *float32
+	Inventory *int
+}
+
+func (s *Service) CreateBook(bookEntry EntryBookRequest) (Book, error) {
+	createdAt := time.Now().UTC().Round(time.Millisecond) //Atribute creating and updating time to the new entry. UpdateAt can change later.
+	newBook := Book{
+		ID:        uuid.New(), //Atribute an ID to the entry
+		Name:      bookEntry.Name,
+		Price:     bookEntry.Price,
+		Inventory: bookEntry.Inventory,
+		CreatedAt: createdAt,
+		UpdatedAt: createdAt,
+		//Archived is set to false by defalut inside database
+	}
+	return s.repo.CreateBook(newBook)
+}
+
+func (s *Service) UpdateBook(bookEntry EntryBookRequest, id uuid.UUID) (Book, error) {
```

Comment:
This is weird, there is no reason to pass the `uuid` separately, this looks like you trying to use the same type for the create and update methods. Just write a new type for the UpdateBook request.
```suggestion
type UpdateBookRequest struct {
        ID        uuid.UUID
	Name      string
	Price     *float32
	Inventory *int
}

func (s *Service) UpdateBook(req UpdateBookRequest) (Book, error) {
Comment

Reply:
Yes, I was exactly trying to use the same type for both methods. Thas why I've called the type EntryBookRequest. But I will separate them. No problem.

Reply:
I thought is was no problem, but I'm facing some of them, since we have different structs to Update and Create. The function `FilledFields`, that is defined at `service.go`, now has to deal with this different types, and I'm having to write a lot of code just to ajust the incoming the book data to pass it to this little function. It makes more sense to me if we transfer this function (or its functionality) to the http layer. It could be a function that verifies the fields and already copy them to the proper struct (CreateBookResquest or UpdateBookRequest).
Ooooor we can come back to having just one type for both methods.

Reply:
You should have separate functions to validate each one of the structs, also you can move it to the handlers layer if you want to.